group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -806,7 +809,29 @@ public void validateOffsetCommit(
         if (memberEpoch < 0 && members().isEmpty()) return;
 
         final ConsumerGroupMember member = getOrMaybeCreateMember(memberId, false);
-        validateMemberEpoch(memberEpoch, member.memberEpoch());
+        if (member.useClassicProtocol()) {
+            validateMemberInstanceId(member, groupInstanceId);
```

Comment:
When a member is fenced, it is removed from the group. So in this particular case, `getOrMaybeCreateMember` will throw a `UnknownMemberIdException` exception before we have a change to validate the instance id here. If we want to validate it, it should be done before, I think.

Reply:
Let's remove this validation because we have to do `getOrMaybeCreateMember(memberId, false);` and know whether the member uses the classic protocol before deciding whether instance id should be validated.